Runtime/Data/Collection/ObservableList.cs

```diff
@@ -85,16 +187,45 @@ public virtual void RemoveFirst(TElement element)
         /// </summary>
         /// <param name="element">The element to remove.</param>
         [RequiresBehaviourState]
-        public virtual void RemoveLast(TElement element)
+        public virtual void RemoveLastOccurrence(TElement element)
         {
-            int index = elements.LastIndexOf(element);
+            int index = Elements.LastIndexOf(element);
             if (index == -1)
             {
                 return;
             }
 
-            elements.RemoveAt(index);
-            EmitRemoveEvents(element);
+            RemoveAt(index);
+        }
+
+        /// <summary>
+        /// Removes an element at the given index from the collection.
+        /// </summary>
+        /// <remarks>
+        /// Allows the use of a wrapped and clamped index to prevent indices being out of bounds and doing negative queries such as `-1` sets the last element.
+        /// </remarks>
+        /// <param name="index">The index to remove at.</param>
```

Comment:
```suggestion
        /// <param name="index">The index to remove at. In case this index is out of bounds for the collection it will wrap around.</param>
```